### PR TITLE
remove unused bdd code

### PR DESCRIPTION
**Pull request by arifogel**
This change is [Reviewable](https://reviewable.io/reviews/batfish/batfish/743)

### CODE REVIEW

**Comment by dhalperi** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[projects/batfish/src/main/java/org/batfish/symbolic/abstraction/Abstraction.java, line 124 at r1](https://reviewable.io:443/reviews/batfish/batfish/743#-L0BNL8-_15BzmycQY8W:-L0BNL8-_15BzmycQY8X:b-rsu0x4) ([raw file](https://github.com/batfish/batfish/blob/0220d6c0c834469a34dbb099e03bd1352ca9d505/projects/batfish/src/main/java/org/batfish/symbolic/abstraction/Abstraction.java#L124)):*
> ```Java
>     }
>     for (Entry<Set<String>, List<Prefix>> entry : destinationMap.entrySet()) {
>       Set<String> devices = entry.getKey();
> ```

If you're doing this, simplify to to `valueSet` and remove the commented-out println below.

Perhaps more likely this could be logged at a debug/trace level.

---

*[projects/batfish/src/main/java/org/batfish/symbolic/bdd/BDDRoute.java, line 57 at r1](https://reviewable.io:443/reviews/batfish/batfish/743#-L0BNbd9fIVjPH1RMtml:-L0BNbd9fIVjPH1RMtmm:b-slxujf) ([raw file](https://github.com/batfish/batfish/blob/0220d6c0c834469a34dbb099e03bd1352ca9d505/projects/batfish/src/main/java/org/batfish/symbolic/bdd/BDDRoute.java#L57)):*
> ```Java
>     CallbackHandler handler = new CallbackHandler();
>     try {
>       Method m = handler.getClass().getDeclaredMethod("handle", (Class<?>[]) null);
> ```

I'll let Ryan comment on this one?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/743)*

**Comment by arifogel** [comment]
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[projects/batfish/src/main/java/org/batfish/symbolic/abstraction/Abstraction.java, line 124 at r1](https://reviewable.io:443/reviews/batfish/batfish/743#-L0BNL8-_15BzmycQY8W:-L0BXHoePYITP3DZL2HD:b-896fix) ([raw file](https://github.com/batfish/batfish/blob/0220d6c0c834469a34dbb099e03bd1352ca9d505/projects/batfish/src/main/java/org/batfish/symbolic/abstraction/Abstraction.java#L124)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

If you're doing this, simplify to to `valueSet` and remove the commented-out println below.

Perhaps more likely this could be logged at a debug/trace level.
</blockquote></details>

Done.

---

*[projects/batfish/src/main/java/org/batfish/symbolic/bdd/BDDRoute.java, line 57 at r1](https://reviewable.io:443/reviews/batfish/batfish/743#-L0BNbd9fIVjPH1RMtml:-L0BSV8Qrnk1masFKZC3:bh36ao8) ([raw file](https://github.com/batfish/batfish/blob/0220d6c0c834469a34dbb099e03bd1352ca9d505/projects/batfish/src/main/java/org/batfish/symbolic/bdd/BDDRoute.java#L57)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

I'll let Ryan comment on this one?
</blockquote></details>

Ryan? (Commenting to remove from red)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/743)*

File: projects/batfish/src/main/java/org/batfish/symbolic/abstraction/Abstraction.java [hunk 1]
```diff
@@ -120,11 +120,8 @@ private void addCatchAllCase(
     for (Prefix pfx : notDstIps) {
       catchAll.getNotDstIps().add(new IpWildcard(pfx));
     }
-    for (Entry<Set<String>, List<Prefix>> entry : destinationMap.entrySet()) {
-      Set<String> devices = entry.getKey();
-      List<Prefix> prefixes = entry.getValue();
+    for (List<Prefix> prefixes : destinationMap.values()) {
```
Comment (rabeckett) [inline projects/batfish/src/main/java/org/batfish/symbolic/abstraction/Abstraction.java]: This looks fine to me.

File: projects/batfish/src/main/java/org/batfish/symbolic/bdd/BDDRoute.java [hunk 2]
```diff
@@ -52,19 +51,13 @@
     allProtos.add(Protocol.OSPF);
     allProtos.add(Protocol.BGP);
 
-    CallbackHandler handler = new CallbackHandler();
```
Comment (rabeckett) [inline projects/batfish/src/main/java/org/batfish/symbolic/bdd/BDDRoute.java]: The JavaBDD library prints garbage collection information to stdout by default. The only way to suppress this printing is to register a callback method using reflection, which is what this is doing. I think the better solution if you are getting an unused variable warning, would be to uncomment the registerGCCallback line to have no printing by default.

**Comment by rabeckett** [comment]
Reviewed 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[projects/batfish/src/main/java/org/batfish/symbolic/abstraction/Abstraction.java, line 124 at r1](https://reviewable.io:443/reviews/batfish/batfish/743#-L0BNL8-_15BzmycQY8W:-L0C7nWfew5NA89-QqUH:bsrfkwe) ([raw file](https://github.com/batfish/batfish/blob/0220d6c0c834469a34dbb099e03bd1352ca9d505/projects/batfish/src/main/java/org/batfish/symbolic/abstraction/Abstraction.java#L124)):*
<details><summary><i>Previously, arifogel (Ari Fogel) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Looks good

---

*[projects/batfish/src/main/java/org/batfish/symbolic/bdd/BDDRoute.java, line 57 at r1](https://reviewable.io:443/reviews/batfish/batfish/743#-L0BNbd9fIVjPH1RMtml:-L0C7pN1Kg4xZZN83cN3:b-vs1pr4) ([raw file](https://github.com/batfish/batfish/blob/0220d6c0c834469a34dbb099e03bd1352ca9d505/projects/batfish/src/main/java/org/batfish/symbolic/bdd/BDDRoute.java#L57)):*
<details><summary><i>Previously, arifogel (Ari Fogel) wrote…</i></summary><blockquote>

Ryan? (Commenting to remove from red)
</blockquote></details>

Looks good.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/743)*

**Comment by rabeckett** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/743#-:-L0C94HXAJApIJwYG9uG:bnfp4nl)*

**Pull request merged**